packages/hoppscotch-app/helpers/curl/sub_helpers/contentParser.ts

```diff
@@ -0,0 +1,322 @@
+import { HoppRESTReqBody } from "@hoppscotch/data"
+import * as O from "fp-ts/Option"
+import * as RA from "fp-ts/ReadonlyArray"
+import * as S from "fp-ts/string"
+import { pipe, flow } from "fp-ts/function"
+import { tupleToRecord } from "~/helpers/functional/record"
+import { safeParseJSON } from "~/helpers/functional/json"
+
+const contentTypeIs =
+  (cType: HoppRESTReqBody["contentType"]) => (input: O.Option<unknown>) =>
+    pipe(
+      input,
+      O.match(
+        () => null,
+        () => cType
+      ),
+      O.fromNullable
+    )
+
+const isJSON = flow(safeParseJSON, contentTypeIs("application/json"))
+
+const isXML = (rawData: string) =>
```

Comment:
Functions like `isJSON` have return type of option. They return `O.some(contentType)` or `O.none`, so it made sense to name them `isJSON`.

Reply:
Exactly, if I hide the function body and just glance through the function, will you expect this function to return a boolean. This is standard we follow for almost all the `isXXX` functions in our codebase or generally in any codebase. Instead of writing things like this,  you can just make these series of functions return a boolean and then write a function like this.

```ts
const getContentTypeFromRawData = (rawData: string) => {
  if (isJSON(rawData)) return O.some("application/json")
  else if (isXML(rawData)) return O.some("application/xml")
  else if (isHTML(rawData)) return O.some("text/html")
  // And so on....

  // Nothing matched
  return O.none
}
```

This also changes the `detectContentType` to be like this
```ts
pipe(
    getContentTypeFromRawData(rawData),
    O.getOrElse(() =>
      pipe(
        rawData,
```

Reply:
Never break expectations of how a function and a type works and a purpose is just to look pretty. It just creates more confusion and chaos in the long term than you expect.

Won't you be spooked if you saw something which looked like a duck and it suddenly started barking ?

Reply:
> Won't you be spooked if you saw something which looked like a duck and it suddenly started barking ?

:slightly_smiling_face:

Reply:
May I rename the functions instead of refactoring the lot? Like renaming the `isJSON` function to `checkJSONData`?
Since we are dealing with options all over the place in these functions, it might not be the best thing to convert them to boolean and then back to `option`.

Reply:
I removed the option and used plain if-else inside detectContentType function itself